### ISSUE TITLE
[1.0.0] Cannot use stateful widget and hooks together

### ISSUE THREAD

**Issue by itsJoKr** [rrousselGit/river_pod#772]
Hey. I have this pretty big issue for me.

Before (0.14.0) I was able to use Hooks + Stateful widget. But with 1.0.0, there is no useProvider hook, but I need to use Consumer. Not sure what is the reason behind this change.

Problem is that I cannot use my hook + consumer + Stateful widget. I can use:
- `StatefulHookWidget ` but I will be missing ref.watch.
- `ConsumerStatefulWidget ` where I have ref.watch but will be missing hooks.

**Describe the solution you'd like**
Either revert the `useProvider` hook or allow us to use hooks inside `ConsumerState`.

**Comment by matuella** [comment]
I think https://github.com/rrousselGit/river_pod/issues/763 answers your problems with using alongside hooks.

 TLDR: use `HookConsumerWidget`, apparently there won't be a similar syntax to `useProvider` in `>=1.0.0`.

**Comment by matuella** [comment]
I think https://github.com/rrousselGit/river_pod/issues/763 answers your problems with using alongside hooks.

 TLDR: use `HookConsumerWidget`, apparently there won't be a similar syntax to `useProvider` in `>=1.0.0`.

**Comment by rrousselGit** [comment]
You can use `ConsumerStatefulWidget` + `HookBuilder`

**Comment by rrousselGit** [comment]
I have added a `StatefulHookConsumerWidget`. It's not relased yet but is available on `master`. So I'll close this for now

**Comment by Shiba-Kar** [comment]
love your work man !!! @rrousselGit
